nbs/methods.ipynb

```diff
@@ -1131,6 +1229,75 @@
     "        )"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| hide\n",
+    "for method in [\"forecast_proportions\", \"average_proportions\", \"proportion_averages\"]:\n",
+    "    cls_middle_out = MiddleOutSparse(middle_level=\"level2\", top_down_method=method)\n",
+    "    if cls_middle_out.insample:\n",
+    "        assert method in [\"average_proportions\", \"proportion_averages\"]\n",
+    "        test_close(\n",
+    "            cls_middle_out(\n",
+    "                S=S,\n",
+    "                y_hat=S @ y_hat_bottom,\n",
+    "                y_insample=S @ y_bottom,\n",
+    "                tags=tags,\n",
+    "            )[\"mean\"],\n",
+    "            S @ y_hat_bottom,\n",
+    "        )\n",
+    "    else:\n",
+    "        test_close(\n",
+    "            cls_middle_out(S=S, y_hat=S @ y_hat_bottom, tags=tags)[\"mean\"],\n",
+    "            S @ y_hat_bottom,\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| hide\n",
+    "for method in [\"forecast_proportions\", \"average_proportions\", \"proportion_averages\"]:\n",
+    "    cls_middle_out = MiddleOut(middle_level=\"level2\", top_down_method=method)\n",
+    "    cls_middle_out_sparse = MiddleOutSparse(\n",
+    "        middle_level=\"level2\", top_down_method=method\n",
+    "    )\n",
+    "    if cls_middle_out.insample:\n",
+    "        assert method in [\"average_proportions\", \"proportion_averages\"]\n",
+    "        test_close(\n",
+    "            cls_middle_out(\n",
+    "                S=S,\n",
+    "                y_hat=S @ y_hat_bottom,\n",
+    "                y_insample=S @ y_bottom,\n",
+    "                tags=tags,\n",
+    "            )[\"mean\"],\n",
+    "            cls_middle_out_sparse(\n",
+    "                S=S,\n",
+    "                y_hat=S @ y_hat_bottom,\n",
+    "                y_insample=S @ y_bottom,\n",
+    "                tags=tags,\n",
+    "            )[\"mean\"],\n",
+    "            np.finfo(np.float64).eps,\n",
+    "        )\n",
+    "    else:\n",
+    "        test_close(\n",
+    "            cls_middle_out(S=S, y_hat=S @ y_hat_bottom, tags=tags)[\"mean\"],\n",
+    "            cls_middle_out_sparse(\n",
+    "                S=S,\n",
+    "                y_hat=S @ y_hat_bottom,\n",
+    "                y_insample=S @ y_bottom,\n",
+    "                tags=tags,\n",
+    "            )[\"mean\"],\n",
+    "            np.finfo(np.float64).eps,\n",
+    "        )"
+   ]
+  },
```

Comment:
I forgot this in the previous PR, but I think it's good to move the Sparse versions of reconciliation methods below the tests of their Dense counterparts and also add the cells for generating the docs for the sparse versions (e.g. `show_doc(MiddleOutSparse, title_level=3)`, etc - see what is below the dense counterparts). This way we can generate docs also for the sparse versions, which are now omitted when generating the documentation.

Reply:
That sounds good. I will sort out the docs for all the sparse methods once this PR is wrapped up.